spec/models/user_spec.rb

```diff
@@ -1 +1,82 @@
 require 'rails_helper'
+RSpec.describe User, type: :model do
+  describe 'Valid?' do
+    let(:user) { build(:user) }
+    it 'valid with valid attributes' do
+      expect(user).to be_valid
+    end
+  end
+
+  describe '.filter' do
+    let!(:user_has_first_project) { create_list(:user_has_first_project, 2) }
+    let!(:users) { create(:user) }
+
+    context 'call without param' do
+      it 'returns all users' do
+        expect(User.filter(nil)).to eq User.includes(:project)
+      end
+    end
+
+    context 'when method with id Project.first' do
+      it 'returns users list by Project.first.id' do
+        id = Project.first.id
+        expect(User.filter(id)).to eq equal_for_filter(id)
+      end
+      def equal_for_filter(id)
+        Project.find(id).users.includes(:project)
+      end
+    end
+  end
+
+  describe '.as_json' do
+    context 'when array of instances converting to json' do
+      let!(:users_list) { create_list(:user_has_first_project, 5) }
+      let(:users) { User.includes(:project) }
+
+      it 'returns an array of objects as json' do
+        user_project_info = users.map do |user|
+          user_project_info = user.project
+          {
+            id: user.id,
+            first_name: user.first_name,
+            last_name: user.last_name,
+            email: user.email,
+            phone: user.phone,
+            project: {
+              id: user_project_info.id,
+              name: user_project_info.name,
+              desc: user_project_info.desc
+            }
+          }
+        end
+        expect(User.as_json(users)).to eq user_project_info.to_json
+      end
+    end
+  end
+
+  describe '.from_omniauth' do
+    subject { described_class.from_omniauth(access_token) }
+    context 'when user use Google SSO' do
+      let(:access_token) { OmniAuth.config.mock_auth[:google] }
+      let(:user_valid) { create(:user, email: 'test@google.com') }
+
+      it 'returns created user' do
+        user_valid
+        user = User.find_by(email: access_token[:info][:email])
+        expect(user).to be_valid
+      end
+
+      it 'returns not valid new user' do
+        data = access_token[:info]
+        user = User.new(
+          first_name: data[:name].partition(' ').first,
+          last_name: data[:name].partition(' ').last,
+          email: data[:email], project: Project.first
+        )
+        user[:encrypted_password] = subject[:encrypted_password]
```

Comment:
check if User.count change
Retrieve user by email and check fields (first_name and last_name)